src/caselawclient/search_result.py

```diff
@@ -0,0 +1,264 @@
+            )[0]
+            if node.xpath(
+                "search:extracted/akn:FRBRdate[(@name='judgment' or @name='decision')]/@date",
+                namespaces=namespaces,
+            )
+            else ""
+        )
+        transformation_date = (
+            node.xpath(
+                "search:extracted/akn:FRBRdate[@name='transform']/@date",
+                namespaces=namespaces,
+            )[0]
+            if node.xpath(
+                "search:extracted/akn:FRBRdate[@name='transform']/@date",
+                namespaces=namespaces,
+            )
+            else ""
+        )
+        content_hash = (
+            node.xpath("search:extracted/uk:hash", namespaces=namespaces)[0].text
+            if node.xpath("search:extracted/uk:hash", namespaces=namespaces)
+            else ""
+        )
+
+        try:
+            date = dateparser.parse(date_string)
+        except ParserError as e:
+            logging.warning(
+                f'Unable to parse document date "{date_string}". Full error: {e}'
+            )
+            date = None
+
+        try:
+            court = courts.get_by_code(court_code)
+        except CourtNotFoundException:
+            court = None
+
+        try:
+            return SearchResult(
+                uri=uri,
+                neutral_citation=neutral_citation,
+                name=metadata_name,
+                matches=matches_html,
+                court=court,
+                date=date,
+                content_hash=content_hash,
+                transformation_date=transformation_date,
+                meta=meta,
+            )
+        except caselawclient.errors.MarklogicAPIError as e:
+            logging.warning(
+                f"Unable to create search result for {uri}. Has it been deleted? Full error: {e}"
+            )
+            return None
+
+
+class EditorStatus(Enum):
+    """
+    Enum representing the editor status.
+    """
+
+    HOLD = "hold"
+    IN_PROGRESS = "in progress"
+    NEW = "new"
+
+
+@dataclass
+class SearchResultMeta:
+    """
+    Represents the metadata of a search result.
+
+    Attributes:
+        author (str): The author of the search result.
+        author_email (str): The email of the author.
+        consignment_reference (str): The consignment reference.
+        submission_datetime (Optional[datetime]): The submission datetime.
+        assigned_to (str): The assigned editor.
+        editor_hold (str): The editor hold status.
+        editor_priority (str): The editor priority.
+        last_modified (str): The last modified date.
+
+    Properties:
+        editor_status (EditorStatus): The editor status based on the metadata.
+    """
+
+    author: str = ""
+    author_email: str = ""
+    consignment_reference: str = ""
+    submission_datetime: Optional[datetime] = None
+    assigned_to: str = ""
+    editor_hold: str = "false"
+    editor_priority: str = ""
+    last_modified: str = ""
+
+    @property
+    def editor_status(
```

Comment:
In EUI need to do `item.meta.editor_status.value` instead of `item.meta.editor_status.value`